src/lib/utils.ts

```diff
@@ -29,3 +30,64 @@ export const sliderGradient = (
 
   return gradientSteps.map((c) => display(c)).join(', ');
 };
+
+export const hashToStoreValues = (
+  hash: string,
+): {
+  bg: PlainColorObject;
+  fg: PlainColorObject;
+  format: ColorFormatId;
+} | void => {
+  if (hash === '') return;
+  hash = decodeURIComponent(hash);
+
+  const [formatValue, bgValue, fgValue] = hash.split('__') as [
+    string,
+    string,
+    string,
+  ];
+  if (!bgValue || !fgValue) return;
+
+  function decodeColor(colorHash: string) {
+    colorHash = colorHash.replaceAll('_', ' ');
+    colorHash = colorHash.replaceAll('~', '%');
+    return parse(colorHash);
```

Comment:
I think we can simplify this function a bit (and handle the gamut-mapping question) by using `to()` here instead of `parse()`:

```suggestion
    return to(colorHash, format, { inGamut: true });
```

That requires moving the `if (!FORMATS.includes(...))` check up earlier (which might be clearer anyway), but then below we can get rid of the manual object creation:

```ts
  const bg = decodeColor(bgValue);
  if (!bg) return;
  const fg = decodeColor(fgValue);
  if (!fg) return;

  return { bg, fg, format };
```